tests: Add tests for processor switch with MultiSim

erin-le opened the pull request:
This PR adds a test that checks if switching processors works when using MultiSim. It tests switching between combinations of Atomic, Timing, and O3 core types by running an X86 kernel boot on one core type, then switching to another core type and running for another 1 billion ticks.

erin-le committed:
tests: add tests for processor switch with multisim

This commit adds tests that switch between KVM and TIMING on X86
NPB size S workloads using multisim.

erin-le committed:
tests: update tests for multisim processor switching

The previous commit tests processor switching with multisim
using processor.switch(), which only switches the processor
of the last simulation. This commit updates the test to use
simulator.switch_processor(), which correctly switches the
processors of all simulations.

erin-le committed:
tests: make changes to processor switch multisim tests

This commit removes an extra config file from the processor
switch multisim tests. It also modifies the test to only run up
until kernel boot before switching processors, and reduces the
number of jobs. It also changes valid_hosts to
constants.supported_hosts.

erin-le committed:
tests: change local filepath to obtain_resource call

erin-le committed:
tests: add copyright

erin-le committed:
tests: add readme for multisim tests

Harshil2107 committed:
Merge branch 'develop' into multisim-tests-processor-switch

erin-le committed:
tests: clean up multisim processor switch tests

erin-le committed:
tests: Modify MultiSim processor switching test

This commit modifies the MultiSim processor switching test to try
switching between different combinations of the ATOMIC, TIMING,
and O3 cpus, instead of essentially running the same test three
times.

erin-le committed:
tests: change test script name for multisim processor switching

erin-le committed:
Merge branch 'develop' into multisim-tests-processor-switch

erin-le committed:
tests: update README for multisim tests

erin-le committed:
tests: remove debugging print statement

powerjg commented on tests/gem5/multisim/configs/x86-processor-switch.py:
```diff
@@ -0,0 +1,99 @@
+        print("Dumping and resetting stats at kernel boot! Hypercall 1")
+        print("Switching processors at kernel boot! Hypercall 1")
+        simulator.switch_processor()
+        simulator.set_hypercall_absolute_max_ticks(1_000_000)
```
Can we run a little longer? Maybe `1_000_000_000`?

powerjg reviewed (commented):
One tiny thing. Otherwise, this is great.

erin-le committed:
tests: run for 1 billion ticks after switching processors

This commit modifies the test script to run for 1 billion ticks
instead of 1 million ticks after switching processors.

erin-le committed:
tests: use relative max ticks

This commit changes the processor switching multisim tests to
use relative max ticks instead of absolute max ticks, as we want
to run for 1 billion ticks after switching processors, rather than
1 billion ticks from the start of simulation.

erin-le committed:
tests: update multisim tests to run to try to avoid conflicts

erin-le committed:
Merge branch 'develop' into multisim-tests-processor-switch

erin-le committed:
Merge branch 'develop' into multisim-tests-processor-switch
